fontbe/src/glyphs.rs

```diff
@@ -123,24 +123,51 @@ impl Work<Context, Error> for GlyphWork {
         // Hopefully in time https://github.com/harfbuzz/boring-expansion-spec means we can drop this
         let glyph = cubics_to_quadratics(glyph);
 
-        // TODO refine (submodel) var model if glyph locations is a subset of var model locations
-
         match glyph {
             CheckedGlyph::Composite { name, components } => {
                 let composite = create_composite(context, &name, default_location, &components)?;
                 context.set_glyph(name, composite.into());
             }
-            CheckedGlyph::Contour { name, contours } => {
-                // Draw the default outline of our simple glyph
-                let Some(path) = contours.get(default_location) else {
+            CheckedGlyph::Contour { name, paths } => {
+                // Convert paths to SimpleGlyph so we can get consistent point streams
+                let paths = paths
+                    .into_iter()
+                    .map(|(loc, path)| {
+                        match SimpleGlyph::from_kurbo(&path).map_err(|e| Error::KurboError {
+                            glyph_name: self.glyph_name.clone(),
+                            kurbo_problem: e,
+                            context: path.to_svg(),
+                        }) {
+                            Ok(glyph) => Ok((loc, glyph)),
+                            Err(e) => Err(e),
+                        }
+                    })
+                    .collect::<Result<HashMap<_, _>, Error>>()?;
+
+                // Establish the default outline of our simple glyph
+                let Some(base_glyph) = paths.get(default_location) else {
                     return Err(Error::GlyphError(ir_glyph.name.clone(), GlyphProblem::MissingDefault));
                 };
-                let base_glyph = SimpleGlyph::from_kurbo(path).map_err(|e| Error::KurboError {
-                    glyph_name: self.glyph_name.clone(),
-                    kurbo_problem: e,
-                    context: path.to_svg(),
-                })?;
-                context.set_glyph(name, base_glyph.into());
+                context.set_glyph(name.clone(), base_glyph.clone().into());
+
+                let point_seqs: HashMap<_, _> = paths
+                    .into_iter()
+                    .map(|(loc, glyph)| {
+                        let points = glyph
+                            .contours()
+                            .iter()
+                            .flat_map(|c| c.iter())
+                            .map(|cp| Point::new(cp.x as f64, cp.y as f64))
+                            .collect();
+                        (loc, points)
+                    })
+                    .collect();
+
+                let deltas = var_model
+                    .deltas(&point_seqs)
```

Comment:
(I thought I had commented on this but it got lost, I'll rephrase)

We want to pass the phantom points to the var_model to compute deltas for them.

Also before computing deltas, we want to check that the on/off-curve flags are identical for simple glyphs; for composites, the component glyph name and the 2x2 transform mast be the same (only x,y offset can vary).
Fonttools currently skips and issues a warning for incompatible glyphs, we might do the same.
https://github.com/fonttools/fonttools/blob/3b9a73ff8379ab49d3ce35aaaaf04b3a7d9d1655/Lib/fontTools/varLib/__init__.py#L276-L280

Reply:
not just the on/off curve flags, actually, but also the number of contours and the list of end point indices for each contour. Basically what fontTools calls _GlyphControls here:
https://github.com/fonttools/fonttools/blob/3b9a73ff8379ab49d3ce35aaaaf04b3a7d9d1655/Lib/fontTools/ttLib/tables/_g_l_y_f.py#L375-L383

Reply:
> We want to pass the phantom points to the var_model to compute deltas for them.

Done, albeit using a dumbed down phantom calculator.

> before computing deltas, we want to ...

Good point, not yet done.

Reply:
Punted to https://github.com/googlefonts/fontmake-rs/issues/241